repos/re-theme/src/context/platform.native.js

```diff
@@ -0,0 +1,37 @@
+let RNPlatform
+
+/**
+ * Loads the react-native platform inside a function call
+ * Ensures react-native has been initialized before calling
+ * @function
+ * 
+ * @return {Object} - Platform object to use when building the theme
+ */
+const loadReactNativePlatform = () => {
+  const { Platform } = require('react-native')
+  RNPlatform = Platform
```

Comment:
* The `require` cache we talked about was for node.
* This code should only ever run within react-native and  never on node, so it doesn't work the same way.
* Basically the `metro` bundler polyfills the global `require` method
  * I'm not 100% sure if the cache works the same way for at all within the `metro` environment
  * I do know that in a dev env, the `metro#require` actually makes a call to the backend server to load the content
  * This caches that call so it's only made once ( again, this is in a dev environment )
  * In production all require calls are run at built time and actually included with the bundle
    * At this point I have no idea how the `require` call  is resolved, so this just ensures is only run once

I think there would be some value in digging deeper into how `require` works within metro
  * There's this react-native thing call `ram-bundler`, that basically makes require calls dynamic
  * I have not tried it, but it may work in this case
  * It's definitely an area I plan to investigate more when I have the oppertunity

Reply:
ah right, node. Alright makes sense.